File: src/main/java/com/dh/ondot/schedule/domain/enums/AlarmMode.java
package com.dh.ondot.schedule.domain.enums;

import com.dh.ondot.core.exception.ErrorCode;
import com.dh.ondot.core.exception.UnsupportedException;

import java.util.Locale;

public enum AlarmMode {
    SILENT,
    VIBRATE,
    SOUND;

    public static AlarmMode from(String mode) {
        try {
            return AlarmMode.valueOf(mode.toUpperCase(Locale.ENGLISH));
        } catch (IllegalArgumentException e) {
            throw new UnsupportedException (ErrorCode.UNSUPPORTED_ALARM_MODE, mode);
        }
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/enums/AlarmMode.kt
package com.dh.ondot.schedule.domain.enums

import com.dh.ondot.core.exception.ErrorCode
import com.dh.ondot.core.exception.UnsupportedException
import java.util.Locale

enum class AlarmMode {
    SILENT,
    VIBRATE,
    SOUND;

    companion object {
        @JvmStatic
        fun from(mode: String): AlarmMode =
            try {
                valueOf(mode.uppercase(Locale.ENGLISH))
            } catch (e: IllegalArgumentException) {
                throw UnsupportedException(ErrorCode.UNSUPPORTED_ALARM_MODE, mode)
            }
    }
}
